website/static/js/tests/share/stats.test.js

```diff
@@ -0,0 +1,231 @@
+                      'min_doc_count': 0
+                    }
+                  }
+                },
+                'filters': {}
+              };
+            assert.deepEqual(returnedAgg, requiredAgg);
+        });
+    });
+
+    describe('#sourcesByDatesAgg', () => {
+
+        it('builds correct query, aggregation and filters for a sourcesByDates aggregation', () => {
+            var returnedAgg = Stats.sourcesByDatesAgg();
+            var requiredAgg = {
+                'aggregations': {
+                  'sourcesByTimes': {
+                    'terms': {
+                      'field': '_type',
+                      'size': 0,
+                      'exclude': 'of|and|or',
+                      'min_doc_count': 0
+                    },
+                    'aggregations': {
+                      'articlesOverTime': {
+                          'aggregations': {
+                              'articlesOverTime': {
+                                'date_histogram': {
+                                    'field': 'providerUpdatedDateTime',
+                                    'interval': 'week',
+                                    'min_doc_count': 0,
+                                    'extended_bounds': {
+                                        'min': 1429562338929,
+                                        'max': 1437424738929
+                                    }
+                                }
+                              }
+                            },
+                            'filter': {
+                                'range': {
+                                    'providerUpdatedDateTime': {
+                                        'gte': 1429562338929,
+                                        'lte': 1437424738929
+                                    }
+                                }
+                            }
+                        }
+                    }
+                  }
+                },
+            };
+
+
+            requiredAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.aggregations.articlesOverTime.date_histogram.extended_bounds = returnedAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.aggregations.articlesOverTime.date_histogram.extended_bounds;
+            requiredAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.filter.range.providerUpdatedDateTime = returnedAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.filter.range.providerUpdatedDateTime ;
+            var diffInMs = requiredAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.aggregations.articlesOverTime.date_histogram.extended_bounds.max - returnedAgg.aggregations.sourcesByTimes.aggregations.articlesOverTime.aggregations.articlesOverTime.date_histogram.extended_bounds.min;
+            var testDate = new Date(0);
+            testDate.setUTCSeconds(diffInMs/1000);
+            assert.equal(testDate.getMonth(),3);
```

Comment:
This isn't actually testing anything about the return value of `sourcesByDatesAgg`. What are you trying to test here?